QuizApp/QuizApp/Login/LoginButton.swift

```diff
@@ -0,0 +1,52 @@
+import UIKit
+
+class LoginButton: UIButton {
+
+    var hasValidInputForEmail = false
```

Comment:
No. :)
Every UI component must be as stupid as possible and has to do only what it is declared to do. So, instead of having these booleans here, you should only have one: `isEnabled`. When you change that field, you should re-style this view.

UIViewController should observe changes from email & password inputs and modify `isEnabled` field in this view.